extensions/ql-vscode/src/contextual/astBuilder.ts

```diff
@@ -95,8 +97,8 @@ export default class AstBuilder {
           const parent = idToItem.get(childToParent.has(id) ? childToParent.get(id)! : -1);
 
           if (parent) {
-            const astItem = item as AstItem;
-            astItem.parent = parent;
+            const astItem = item as unknown as AstItem;
```

Comment:
Bit sad, but guess we're missing the `parent` property. Perhaps there's room to relate `AstItem` and `RootAstItem` so that you can easily convert between them (with/without parent) without having to cast to unknown, but not blocking.

Reply:
Yeah, I'm not happy that I sometimes need to explicitly cast to `RootAstItem` also even when it is not necessarily a root.

Maybe I should rename `AstItem` --> `ChildAstItem` and `RootAstItem` --> `AstItem`.

Reply:
That can work. Or make `parent` optional on `AstItem`, and `RootAstItem` override it to `undefined`.